emukit/model_wrappers/gpy_model_wrappers.py

```diff
@@ -100,6 +100,45 @@ def Y(self) -> np.ndarray:
         """
         return self.model.Y
 
+    def generate_hyperparameters_samples(self, n_samples = 20, n_burnin = 100, subsample_interval  = 10,
+                                         step_size = 1e-1, leapfrog_steps = 20) -> None:
+        """
+        Generates the samples from the hyper-parameters
+        :param n_samples: Number of generated samples.
+        :param n_burning: Number of initial samples not used.
+        :param subsample_interval: Interval of subsampling from HMC samples.
+        :param step_size: Size of the gradient steps in the HMC sampler.
+        :param leapfrog_steps: Number of gradient steps before each Metropolis Hasting step.
+        :return: A numpy array whose rows are samples of the hyper-parameters.
+
+        """
+        self.model.optimize(max_iters=self.n_restarts)
+        self.model.param_array[:] = self.model.param_array * (1.+np.random.randn(self.model.param_array.size)*0.01)
+        hmc = GPy.inference.mcmc.HMC(self.model, stepsize = step_size)
+        samples = hmc.sample(num_samples = n_burnin + n_samples * subsample_interval, hmc_iters = leapfrog_steps)
+        hmc_samples = samples[n_burnin::subsample_interval]
+
+        return hmc_samples
+
+    @property
+    def hyperparameters_samples(self) -> np.ndarray:
+        """
+        :return: An array of shape num_samples x num_hyperparameters
+        """
+        return self.samples
+
+    def fix_model_hyperparameters(self, sample_hyperparameters: np.array) -> None:
```

Comment:
np.array isn't a type you are looking for. `np.ndarray` is

Reply:
done